trezorlib/eos.py

```diff
@@ -0,0 +1,387 @@
+        )
+
+    accounts = []
+    for account in data['accounts']:
+        accounts.append(
+            proto.EosAuthorizationAccount(
+                account=proto.EosPermissionLevel(
+                    actor=name_to_number(account['permission']['actor']),
+                    permission=name_to_number(account['permission']['permission'])
+                ),
+                weight=int(account['weight'])
+            )
+        )
+
+    waits = []
+    for wait in data['waits']:
+        waits.append(
+            proto.EosAuthorizationWait(
+                wait_sec=int(wait['wait_sec']),
+                weight=int(wait['weight'])
+            )
+        )
+
+    return proto.EosAuthorization(
+        threshold=int(data['threshold']),
+        keys=keys,
+        accounts=accounts,
+        waits=waits,
+    )
+
+
+def parse_new_account(data):
+    owner = parse_authorization(data['owner'])
+    active = parse_authorization(data['active'])
+
+    return proto.EosActionNewAccount(
+        creator=name_to_number(data['creator']),
+        name=name_to_number(data['name']),
+        owner=owner,
+        active=active
+    )
+
+
+def parse_unknown(data):
+    return proto.EosActionUnknown(
+        data=binascii.unhexlify(data)
+    )
+
+def parse_action(action):
+    txAction = proto.EosTxActionAck()
+    data = action['data']
+
+    txAction.common = parse_common(action)
+
+    if action['name'] == 'transfer':
+        txAction.transfer = parse_transfer(data)
+    elif action['name'] == 'voteproducer':
+        txAction.vote_producer = parse_vote_producer(data)
+    elif action['name'] == 'buyram':
+        txAction.buy_ram = parse_buy_ram(data)
+    elif action['name'] == 'buyrambytes':
+        txAction.buy_ram_bytes = parse_buy_rambytes(data)
+    elif action['name'] == 'sellram':
+        txAction.sell_ram = parse_sell_ram(data)
+    elif action['name'] == 'delegatebw':
+        txAction.delegate = parse_delegate(data)
+    elif action['name'] == 'undelegatebw':
+        txAction.undelegate = parse_undelegate(data)
+    elif action['name'] == 'refund':
+        txAction.refund = parse_refund(data)
+    elif action['name'] == 'updateauth':
+        txAction.update_auth = parse_updateauth(data)
+    elif action['name'] == 'deleteauth':
+        txAction.delete_auth = parse_deleteauth(data)
+    elif action['name'] == 'linkauth':
+        txAction.link_auth = parse_linkauth(data)
+    elif action['name'] == 'unlinkauth':
+        txAction.unlink_auth = parse_unlinkauth(data)
+    elif action['name'] == 'newaccount':
+        txAction.new_account = parse_new_account(data)
+    else:
+        txAction.unknown = parse_unknown(data)
+
+    return txAction
+
+
+
+def parse_transaction_json(json):
+    tx = type('Transaction', (object,), {})()
+    tx.chain_id = binascii.unhexlify(json['chain_id'])
+
+    body = json['transaction']
+
+    expiration = int(datetime.strptime(body['expiration'], '%Y-%m-%dT%H:%M:%S').strftime("%s"))
```

Comment:
IMO, it would be better if this didn't adjust to the local timezone:

```suggestion
    expiration = int((datetime.strptime(body['expiration'], '%Y-%m-%dT%H:%M:%S') - datetime(1970, 1, 1)).total_seconds())
```

Reply:
In fact, it's necessary to make the testcases timezone agnostic. As-is, these only work in UTC+3.